### PR TITLE
chore(ci): enforce zero trust for third party workflows

### LINKED ISSUE #2207: Remove allow list from untrusted workflows

### PR DESCRIPTION
**Pull request by heitorlessa**
**Issue number:** #2207

## Summary

Remove `allow_list` from workflow that enforces 3rd party actions to be pinned. Addressed shellcheck issues that weren't identified earlier due to being in allow list.

### Changes

> Please provide a summary of what's being changed

### User experience

> Please share what the user experience looks like before and after this change

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [ ] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -23,11 +23,11 @@ jobs:
 
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v3
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab  # v3.5.2
 
       # Initializes the CodeQL tools for scanning.
       - name: Initialize CodeQL
-        uses: github/codeql-action/init@v2
+        uses: github/codeql-action/init@f31a31c052207cc13b328d6295c5b728bb49568c # v2.13.1@v2
```
Comment (leandrodamascena) [inline .github/workflows/codeql-analysis.yml]:
```suggestion
        uses: github/codeql-action/init@f31a31c052207cc13b328d6295c5b728bb49568c # v2.13.1
```

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -36,4 +36,4 @@ jobs:
           # queries: ./path/to/local/query, your-org/your-repo/queries@main
 
       - name: Perform CodeQL Analysis
-        uses: github/codeql-action/analyze@v2
+        uses: github/codeql-action/analyze@f31a31c052207cc13b328d6295c5b728bb49568c # v2.13.1@v2
```
Comment (leandrodamascena) [inline .github/workflows/codeql-analysis.yml]:
```suggestion
        uses: github/codeql-action/analyze@f31a31c052207cc13b328d6295c5b728bb49568c # v2.13.1
```

**Review by leandrodamascena — CHANGES_REQUESTED** [review]
I haven't checked the hash of the images if it matches the version informed, but it all looks great to me.

**Review by leandrodamascena — APPROVED** [review]
image

**Comment by heitorlessa** [comment]
testing nudge before merging 
 > Sent from <a href='https://neat.run?ref=gh_footer'>Neat</a>

**Pull request merged**